src/compose/ports.ts

```diff
@@ -30,7 +30,7 @@ interface PortRange {
 export class PortMap {
 	private ports: PortRange;
 
-	private constructor(portStrOrObj: string | PortRange) {
+	public constructor(portStrOrObj: string | PortRange) {
```

Comment:
This constructor is used a lot by the tests and the easiest way to make them work was to make it public

Reply:
There's a good reason that this is private, and I don't want anybody to think that this is the way to use that throughout the codebase. I'd 100% prefer a `@ts-ignore` or cast to any.

Reply:
done